neqo-transport/src/streams.rs

```diff
@@ -131,6 +131,8 @@ impl Streams {
                 stats.reset_stream += 1;
                 if let (_, Some(rs)) = self.obtain_stream(*stream_id)? {
                     rs.reset(*application_error_code, *final_size)?;
+                } else {
+                    return Err(Error::StreamStateError);
```

Comment:
What if `self.obtain_stream(*stream_id)?` is `(None, None)`, where `stream_id` is a receive stream that has already been reset?

Reply:
I _think_ `obtain_stream` will still return the stream even is it is in state `ResetRecvd`. `RecvStream::reset()` won't do anything with it though.

But good question if there is any case where `obtain_stream` would return `(None, None)` where we shouldn't close the connection.

Reply:
@mxinden see 795933b87f381a4ef6cae78008dbb4341cd6fe12

Reply:
Thanks for the additional test.

> I _think_ `obtain_stream` will still return the stream even is it is in state `ResetRecvd`.

Yes. But what if `cleanup_closed_streams` is called in between receiving the first and the second reset?

https://github.com/mozilla/neqo/blob/6575c551131b167a632be6b8db45b16fd16d50c7/neqo-transport/src/streams.rs#L333-L343

In that case `obtain_stream` should return `(None, None)`, right?

Reply:
Need to check more, but inserting `server.streams.cleanup_closed_streams();` in between processing the two reset frames in `double_reset_stream` does **not** make the test fail.

Reply:
I believe that we won't drop a stream until we've had the RESET_STREAM frame acknowledged.  You have to do a back-and-forth to get that to happen.

Reply:
@martinthomson isn't the logic in question about processing at the side that _receives_ these stream control frames?